zcash_keys/src/address.rs

```diff
@@ -459,6 +470,21 @@ impl Address {
             Address::Tex(_) => None,
         }
     }
+
+    /// Returns the [`Receiver`]s of this address as a vector. In the case that the underlying
+    /// address is the [`Address::Unified`] variant, this is equivalent to
+    /// [`UnifiedAddress::receivers`] in that it does not return unknown receiver data.
+    pub fn receivers(&self) -> Vec<Receiver> {
+        match self {
+            #[cfg(feature = "sapling")]
+            Address::Sapling(addr) => vec![Receiver::Sapling(*addr)],
+            Address::Transparent(addr) => vec![Receiver::Transparent(*addr)],
+            Address::Unified(ua) => ua.receivers(),
+            Address::Tex(addr) => vec![Receiver::Transparent(TransparentAddress::PublicKeyHash(
```

Comment:
Hmm, this is a potentially dangerous method as it type-erases the TEX nature of those addresses, potentially leading to violating MUST requirements of [ZIP 320](https://zips.z.cash/zip-0320#design-considerations-for-senders).

How is this method intended to be used? Is there any reasonable way to adjust the API such that the ZIP 320 requirements are more obvious to the caller? If not, the method needs to at a minimum document the ZIP 320 requirements.

(The same concern applies to `Address::to_transparent_address`, which is used in `get_transparent_receivers` inside `zcash_client_sqlite`. That method should also have the ZIP 320 requirements documented on it.)

Reply:
The intent of this method is that it be used to get relevant addresses to check (mostly the transparent bits) when the wallet is inspecting chain data to detect that a transaction pertains to the wallet. Since there is no dependency in either direction between `zcash_keys` and `zcash_primitives`, there isn't a good way to implement that check internal to one or the other of those crates. I'll update the documentation to reflect that it should not be used for operations related to transaction creation.